File: banking-service/src/main/java/com/banka1/banking/models/Account.java
package com.banka1.banking.models;

import com.banka1.banking.models.helper.AccountStatus;
import com.banka1.banking.models.helper.AccountSubtype;
import com.banka1.banking.models.helper.AccountType;
import com.banka1.banking.models.helper.CurrencyType;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Account {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long ownerID;

    @Column(nullable = false, unique = true)
    private String accountNumber;

    @Column(nullable = false)
    private Double balance;

    @Column(nullable = false)
    private Double reservedBalance;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private AccountType type;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private CurrencyType currencyType;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private AccountSubtype subtype;

    @Column(nullable = false)
    private Long createdDate;

    @Column(nullable = false)
    private Long expirationDate;

    @Column(nullable = false)
    private Double dailyLimit;

    @Column(nullable = false)
    private Double monthlyLimit;

    @Column(nullable = false)
    private Double dailySpent;

    @Column(nullable = false)
    private Double monthlySpent;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private AccountStatus status;

    @Column(nullable = false)
    private Long employeeID;

    @Column(nullable = false)
    private Double monthlyMaintenanceFee;

    @ManyToOne
    @JoinColumn(name = "company_id")
    private Company company;

}

File: trading-service/dto/Account.go
package dto

type UserRequest struct {
	UserId int64 `json:"userId"`
}

type Account struct {
	ID                    int64   `json:"id"`
	OwnerID               int64   `json:"ownerID"`
	AccountNumber         string  `json:"accountNumber"`
	Balance               float64 `json:"balance"`
	ReservedBalance       float64 `json:"reservedBalance"`
	Type                  string  `json:"type"`
	CurrencyType          string  `json:"currencyType"`
	Subtype               string  `json:"subtype"`
	CreatedDate           int64   `json:"createdDate"`
	ExpirationDate        int64   `json:"expirationDate"`
	DailyLimit            float64 `json:"dailyLimit"`
	MonthlyLimit          float64 `json:"monthlyLimit"`
	DailySpent            float64 `json:"dailySpent"`
	MonthlySpent          float64 `json:"monthlySpent"`
	Status                string  `json:"status"`
	EmployeeID            int64   `json:"employeeID"`
	MonthlyMaintenanceFee float64 `json:"monthlyMaintenanceFee"`
}

type UserAccountsResponse struct {
	Accounts []Account `json:"accounts"`
}

type OTCPremiumFeeDTO struct {
	SellerAccountId uint    `json:"sellerAccountId"`
	BuyerAccountId  uint    `json:"buyerAccountId"`
	Amount          float64 `json:"amount"`
}
